2.13.2 backports 1

gsmet opened the pull request:
Please don't merge, I will merge it myself.

geoand committed:
Avoid leaking memory in SseParser

Relates to: #23997

(cherry picked from commit 5d485f0ca5d500f782144ad263410d8ea1ebb3c8)

dependabot[bot] committed:
Bump scala.version from 2.13.8 to 2.13.9

Bumps `scala.version` from 2.13.8 to 2.13.9.

Updates `scala-reflect` from 2.13.8 to 2.13.9
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.8...v2.13.9)

Updates `scala-library` from 2.13.8 to 2.13.9
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.8...v2.13.9)

Updates `scala-compiler` from 2.13.8 to 2.13.9
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.8...v2.13.9)

---
updated-dependencies:
- dependency-name: org.scala-lang:scala-reflect
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.scala-lang:scala-library
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.scala-lang:scala-compiler
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit b036ea19c19f77576996c533f75e245f96fd7f8e)

cescoffier committed:
Update to Netty 4.1.82 and Brotli4J 1.8.0

- Add a substitution to avoid using the BouncyCastle (BC) PEM reader when BC is unavailable.

(cherry picked from commit b7954d17329980baa4d0b1fc7f4fcba686df6d04)

dependabot[bot] committed:
Bump elasticsearch-opensource-components.version from 8.4.1 to 8.4.2

Bumps `elasticsearch-opensource-components.version` from 8.4.1 to 8.4.2.

Updates `elasticsearch-rest-client` from 8.4.1 to 8.4.2
- [Release notes](https://github.com/elastic/elasticsearch/releases)
- [Changelog](https://github.com/elastic/elasticsearch/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elastic/elasticsearch/compare/v8.4.1...v8.4.2)

Updates `elasticsearch-rest-client-sniffer` from 8.4.1 to 8.4.2
- [Release notes](https://github.com/elastic/elasticsearch/releases)
- [Changelog](https://github.com/elastic/elasticsearch/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elastic/elasticsearch/compare/v8.4.1...v8.4.2)

---
updated-dependencies:
- dependency-name: org.elasticsearch.client:elasticsearch-rest-client
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.elasticsearch.client:elasticsearch-rest-client-sniffer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit 99869a93efb332c367413e8d2b4ca0ca6d714ff2)

geoand committed:
Use proper type for serializing Kotlin response

Fixes: #28096
(cherry picked from commit 26f53a7ee7f956cae0083d216b89649876b79527)

Sgitario committed:
Container image registry is ignored when using quarkus-openshift

This commit fixes several issues when using the `quarkus-openshift` extension:
- When providing the registry using the property `quarkus.container-image.image` (for example, quay.io/user/app:version, the registry is `quay.io`), the extension `quarkus-openshift` ignores the registry and uses the fallback property which is `docker.io`.

- The `quarkus-openshift` extension always creates the ImageStream resource regardless of the container image builder used (s2i, docker, jib, ...). However, when the user provides a docker image, it was using it for the container image of the Deployment resource, and this ImageStream was never in use. After these changes, the Deployment resource always uses the ImageStream resource, so no changes in the documentation are necessary.

- When the user provides a docker image and the deployment kind is "DeploymentConfig" (the default option), the `quarkus-openshift` extension removes the trigger image change params, so when deploying the application, the application is not automatically started.

(cherry picked from commit 827b1ab9431927fe545e8929afde442036e8bccf)

sberyozkin committed:
Retry the admin token acquisition in DevServices for Keycloak

(cherry picked from commit f121147ce45f7564c463bd48e80949156ace7690)

sberyozkin committed:
Improve OpenApi CORS message

(cherry picked from commit c78f626346f2d89395d8e29bcd904e8ed4ab1fb1)

dependabot[bot] committed:
Bump mongo-client.version from 4.7.1 to 4.7.2

Bumps `mongo-client.version` from 4.7.1 to 4.7.2.

Updates `mongodb-driver-sync` from 4.7.1 to 4.7.2
- [Release notes](https://github.com/mongodb/mongo-java-driver/releases)
- [Commits](https://github.com/mongodb/mongo-java-driver/compare/r4.7.1...r4.7.2)

Updates `mongodb-driver-core` from 4.7.1 to 4.7.2
- [Release notes](https://github.com/mongodb/mongo-java-driver/releases)
- [Commits](https://github.com/mongodb/mongo-java-driver/compare/r4.7.1...r4.7.2)

Updates `mongodb-driver-legacy` from 4.7.1 to 4.7.2
- [Release notes](https://github.com/mongodb/mongo-java-driver/releases)
- [Commits](https://github.com/mongodb/mongo-java-driver/compare/r4.7.1...r4.7.2)

Updates `mongodb-driver-reactivestreams` from 4.7.1 to 4.7.2
- [Release notes](https://github.com/mongodb/mongo-java-driver/releases)
- [Commits](https://github.com/mongodb/mongo-java-driver/compare/r4.7.1...r4.7.2)

Updates `bson` from 4.7.1 to 4.7.2
- [Release notes](https://github.com/mongodb/mongo-java-driver/releases)
- [Commits](https://github.com/mongodb/mongo-java-driver/compare/r4.7.1...r4.7.2)

---
updated-dependencies:
- dependency-name: org.mongodb:mongodb-driver-sync
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.mongodb:mongodb-driver-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.mongodb:mongodb-driver-legacy
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.mongodb:mongodb-driver-reactivestreams
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.mongodb:bson
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit 4f7610d7fc034f468284e36b973d9e9ebf61a9ec)

aloubyansky committed:
Make sure the app model is initialized from the POMs provided by the
Maven plugin that could be manipulated by Maven extensions

(cherry picked from commit a426f6a372fdfceaaee7d2e3c1342805aa6818df)

yrodiere committed:
Add link to datasource guide for JDBC extensions

(cherry picked from commit c99a10987807e76e86adfbc1b13e925f9c5c562f)

yrodiere committed:
Add link to datasource guide for datasource extension

(cherry picked from commit 6daaa49704c41eacd2147972cfc2ec60e8e88230)

gsmet committed:
Upgrade protobuf to 3.19.6

(cherry picked from commit ed68ec2771fccefe6d59d0b8fe6281f34cfcf8d6)

sberyozkin committed:
Update OIDC DevUI to accept hybrid application types

(cherry picked from commit c540c11c71f67f0c9a57d375f4c9c71878d84b9e)

Karm committed:
Allow Linux extensions on Windows with containers

(cherry picked from commit 00392e75baaa66f0bc5ed9c36e7e25d3d7f66407)

dependabot[bot] committed:
Bump elasticsearch-opensource-components.version from 8.4.2 to 8.4.3

Bumps `elasticsearch-opensource-components.version` from 8.4.2 to 8.4.3.

Updates `elasticsearch-rest-client` from 8.4.2 to 8.4.3
- [Release notes](https://github.com/elastic/elasticsearch/releases)
- [Changelog](https://github.com/elastic/elasticsearch/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elastic/elasticsearch/compare/v8.4.2...v8.4.3)

Updates `elasticsearch-rest-client-sniffer` from 8.4.2 to 8.4.3
- [Release notes](https://github.com/elastic/elasticsearch/releases)
- [Changelog](https://github.com/elastic/elasticsearch/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elastic/elasticsearch/compare/v8.4.2...v8.4.3)

---
updated-dependencies:
- dependency-name: org.elasticsearch.client:elasticsearch-rest-client
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.elasticsearch.client:elasticsearch-rest-client-sniffer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit 02cdd098a382788b0713f90b3f7446f237fda490)

matejvasek committed:
docs: added ':Z' opt to volume mount

Without the option mount won't work on machines with SELinux enabled.

Signed-off-by: Matej Vasek <mvasek@redhat.com>
(cherry picked from commit f50794d0355a2d96c98c269624516d49152d55c2)

geoand committed:
Ensure that RESTEasy Reactive knows the proper return type when Flow is used

(cherry picked from commit 5a2c6a742307226e82d44d1e23ccae8302584b32)

gsmet commented:
@Sgitario I tried to backport these 2 PRs to 2.13 and a test is failing. I'm wondering if I should also backport another one (but we should probably check it's safe enough) or if the test should be adjusted. WDYT?

Sgitario commented:
> @Sgitario I tried to backport these 2 PRs to 2.13 and a test is failing. I'm wondering if I should also backport another one (but we should probably check it's safe enough) or if the test should be adjusted. WDYT?

I'm checking why these two 2 pull requests started failing in 2.13

sberyozkin committed:
Update the keycloak-admin-client doc to show a correct URL for legacy Keycloak

(cherry picked from commit f6d1b7c94e2197b5c1fa9a672bc564d6149115af)

turing85 committed:
Fixed maven and gradle snippet

(cherry picked from commit 7c176d6cbb1e75e9c47d06c9e64abd77056d3d95)

alex-martel committed:
Eventbus documentation uses non existing method

(cherry picked from commit e480f7000758d1a8b994ee7ed4b8406347b4287c)

stuartwdouglas committed:
Fix RR match bug

When matching literals we break out of the wrong loop

(cherry picked from commit 7dfe43aa64c748817c9bf47f669d462279170cd8)

geoand committed:
Provide a dedicated build item for @RegisterProvider instances

(cherry picked from commit 580e126944dc4ba87fafbf1ee1d4f1b6ecff5e6a)

geoand committed:
Introduce a version of @OidcClientFilter for the reactive client

Closes: #16522
(cherry picked from commit 527f84ab052b8d7fc4bc23ff5a5d1ff6cd163027)

geoand committed:
Polish OidcClientRequestReactiveFilter

(cherry picked from commit 6a9327ada2fc68001a7053faaa9fe9b100f74f33)

geoand committed:
Update documentation to display the use of @OidcClientFilter

(cherry picked from commit c46a9b4b5aff217551f63e550d06c4875ba01d07)

michalvavrik committed:
Use fixed ports in Dev Svc for Keycloak on shared network & user request

fixes: #27763
(cherry picked from commit 04582da13da6af81c60047e8600cf256c8196fe3)

geoand committed:
Prevent possible NPE when building violations report

The NPE could happen when Response was used as a return type
and no @Produces annotation was present.

Fixes: https://github.com/quarkusio/quarkus-workshops/issues/183
(cherry picked from commit 1a158520265d48dceb5a39ca5917f2df5d728ec4)

geoand committed:
Ensure that JAX-RS Cookie parameter type is usable in Resource methods

Fixes: #28453
(cherry picked from commit ac731a2c78775dce005c7ee72ff839950435490e)

sberyozkin committed:
Mobe OidcClientFilter annotation to quarkus-oidc-client

(cherry picked from commit 8074d3adc2f1767136b8cc40f9d3ada2c520ba0a)

gsmet committed:
Fix package structure of Kotlin serialization extensions

(cherry picked from commit bf2da28e9b5ab31376312ee536ec5d17b5a7e191)

gsmet committed:
Fix RESTEasy Reactive Kotlin Serialization Common extension

(cherry picked from commit 99969364d2247bb3044c993bde3ac125f59d7b0b)

gsmet committed:
Make sure all CLI commands work with Powershell

Fixe #28465

(cherry picked from commit e5c6af2891e8b391972ee0b5c4b6a189349f8eba)

dependabot[bot] committed:
Bump scala.version from 2.13.9 to 2.13.10

Bumps `scala.version` from 2.13.9 to 2.13.10.

Updates `scala-reflect` from 2.13.9 to 2.13.10
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.9...v2.13.10)

Updates `scala-library` from 2.13.9 to 2.13.10
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.9...v2.13.10)

Updates `scala-compiler` from 2.13.9 to 2.13.10
- [Release notes](https://github.com/scala/scala/releases)
- [Commits](https://github.com/scala/scala/compare/v2.13.9...v2.13.10)

---
updated-dependencies:
- dependency-name: org.scala-lang:scala-reflect
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.scala-lang:scala-library
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.scala-lang:scala-compiler
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit 23a000512c931a6e9c69f8938d6299c930c18379)

geoand committed:
Fix RESTEasy Reactive testsuite

The failure showed up in a few PRs
(like #28429 and #27526)

(cherry picked from commit fbd8fff26634db287b78bcb553bb3acefbca9187)

phillip-kruger committed:
Add Rest Client Guide to Dev UI

(cherry picked from commit 4e96d33c0f17908220d3752e8625ea1e7b8dc4de)
